src/base/bittorrent/torrentimpl.cpp

```diff
@@ -78,6 +78,16 @@ namespace
         return entry;
     }
 
+    QDateTime fromLTTimePoint32(const lt::time_point32 &timePoint)
+    {
+        using namespace std::chrono;
+
+        const auto ltNow = time_point_cast<lt::seconds32>(lt::clock_type::now());
+        const auto secsSinceNow = duration_cast<seconds>(timePoint - ltNow).count();
+        const auto qNow = QDateTime::currentDateTime();
```

Comment:
I would try to avoid calling both `QDateTime::currentDateTime()` and `lt::clock_type::now()`. There are system calls within and should be minimized.
Perhaps `QDateTime::fromStdTimePoint()` and its overloads can help here?

Reply:
> Perhaps `QDateTime::fromStdTimePoint()` and its overloads can help here?

Unfortunately it is unavailable with the compiler I use currently on Linux.
I'll try to figure out if there is an acceptable opportunity to update it.

Reply:
Unfortunately I can't use `QDateTime::fromStdTimePoint()` with MSVC2022 too.

Reply:
>Unfortunately I can't use QDateTime::fromStdTimePoint() with MSVC2022 too.

Probably because the clocks isn't compatible.

Please try this:
```c++
#include "libtorrent/aux_/time.hpp"

QDateTime fromLTTimePoint32(const lt::time_point32 &timePoint)
{
    const std::time_t seconds = lt::aux::to_time_t(timePoint);
    return QDateTime::fromSecsSinceEpoch(seconds);
}
```

Reply:
`lt::aux::to_time_t()` isn't exported in regular builds. Anyway there are two `now()` there too so what's the point?